app/script/calling/CallingRepository.coffee

```diff
@@ -0,0 +1,177 @@
+#
+# Wire
+# Copyright (C) 2016 Wire Swiss GmbH
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see http://www.gnu.org/licenses/.
+#
+
+window.z ?= {}
+z.calling ?= {}
+
+CALLING_CONFIG =
+  CONFIG_UPDATE_INTERVAL: 30 * 60 * 1000 # 30 minutes
+
+# Call repository for all calling interactions.
+class z.calling.CallingRepository
+  ###
+  Extended check for calling support of browser.
+  @return [Boolean] True if calling is supported
+  ###
+  @supports_calling: ->
+    return z.util.Environment.browser.supports.calling
+
+  ###
+  Extended check for screen sharing support of browser.
+  @return [Boolean] True if screen sharing is supported
+  ###
+  @supports_screen_sharing: ->
+    return z.util.Environment.browser.supports.screen_sharing
+
+  ###
+  Construct a new Calling repository.
+
+  @param call_service [z.calling.belfry.CallService] Backend REST API call service implementation
+  @param calling_service [z.calling.CallingService] Backend REST API calling service implementation
+  @param conversation_repository [z.conversation.ConversationRepository] Repository for conversation interactions
+  @param media_repository [z.media.MediaRepository] Repository for media interactions
+  @param user_repository [z.user.UserRepository] Repository for all user and connection interactions
+  ###
+  constructor: (@call_service, @calling_service, @conversation_repository, @media_repository, @user_repository) ->
+    @logger = new z.util.Logger 'z.calling.CallingRepository', z.config.LOGGER.OPTIONS
+
+    @calling_config = ko.observable()
+    @use_v3_api = undefined
+
+    @call_center = new z.calling.belfry.CallCenter @calling_config, @call_service, @conversation_repository, @media_repository, @user_repository
+    @e_call_center = new z.calling.e_call.ECallCenter @calling_config, @conversation_repository, @media_repository, @user_repository
+
+    @calls = ko.pureComputed =>
+      return @call_center.calls().concat @e_call_center.e_calls()
+    @joined_call = ko.pureComputed =>
+      return @e_call_center.joined_e_call() if @e_call_center.joined_e_call()
+      return @call_center.joined_call() if @call_center.joined_call()
```

Comment:
`return @e_call_center.joined_e_call() or @call_center.joined_call()` should do the same